### PR TITLE
Prevent default clicking on markdown preview links when using `openLinksInNewTab`

### PR DESCRIPTION
**Pull request by marywhite**
Describe your changes here.

Closes https://github.com/primer/react/issues/2717

`MarkdownViewer` accepts an `openLinksInNewTab` prop to ensure link previews open in a new tab. Because default is not prevented, navigation is also happening in the working tab.

I've also added gone ahead and added `noopener noreferrer` to the call to `window.open` s/o @mattcosta7 

### Screenshots

Before:

https://user-images.githubusercontent.com/5373900/208768433-ca2095a9-9965-42eb-860c-bb27a2667727.mov

After:

https://user-images.githubusercontent.com/5373900/208768491-87f39bc8-43af-4f18-b00f-92d62903adb6.mov

### Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: src/drafts/MarkdownEditor/MarkdownEditor.test.tsx [hunk 1]
```diff
@@ -800,19 +798,8 @@ describe('MarkdownEditor', () => {
         const link = await waitFor(() => within(getPreview()).getByText('Link'))
 
         await user.click(link)
-        // Note: navigation is not implemented in JSDOM and will log out an
-        // error when clicking the link above. The spy here captures this error
-        // and will assert that it is called only once, otherwise another error
-        // in this test has occurred
-        expect(spy).toHaveBeenCalledTimes(1)
```
Comment (marywhite) [inline src/drafts/MarkdownEditor/MarkdownEditor.test.tsx:807]: This test was previously asserting the buggy behavior - now that we're not navigating the page (the call to window.open is mocked), we don't see any error.

**Review by dusave — APPROVED** [review]

**Review by colebemis — APPROVED** [review]
Thank you 💖

**Review by mattcosta7 — APPROVED** [review]

**Comment by marywhite** [comment]
@joshblack are you able to reenable auto-merge? pushed out a linting fix and I don't have access to merge 🙏

**Comment by joshblack** [comment]
@marywhite on it!

**Pull request merged**